packages/api/tests/endpoints/test-executions.js

```diff
@@ -146,6 +329,17 @@ test('executions can be filtered by workflow', async (t) => {
   t.is(fakeExecutions[1].arn, results[0].arn);
 });
 
+test('GET executions with asyncOperationId filter returns the correct executions', async (t) => {
+  const response = await request(app)
+    .get('/executions?asyncOperationId=0fe6317a-233c-4f19-a551-f0f76071402f')
```

Comment:
I think this is a copy of an existing test, but I don't understand how it works. I don't see where in the setup we ensure that executions exist that reference this async operation

Reply:
Correct this is a copy of an existing test. We're creating an execution with a that asyncOperationId on line 119 and then indexing the array of executions. 

```
  await Promise.all(
    fakeExecutions.map((i) =>
      executionModel
        .create(i)
        .then((record) =>
          indexer.indexExecution(esClient, record, t.context.esAlias)))
  );
```